### PR TITLE
Add java websocket support

### PR DESCRIPTION
**Pull request by zhaomingwork**
A multiple threads websocket server and a multiple threads client for test.

### CODE REVIEW

**Comment by zhaomingwork** [comment]
I think it is OK, pls review when have time. @csukuangfj

File: java-api-examples/src/websocketsrv/AsrWebsocketServer.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ * AsrWebSocketServer has three threads pools, one pool for network io, one pool for asr stream and
+ * one pool for asr decoder.
+ */
+public class AsrWebsocketServer extends WebSocketServer {
+  private static final Logger logger = LoggerFactory.getLogger(AsrWebsocketServer.class);
+  //  Queue between io network io thread pool and stream thread pool, use websocket as the key
+  static LinkedBlockingQueue<WebSocket> streamQueue = new LinkedBlockingQueue<WebSocket>();
```
Comment (csukuangfj) [inline java-api-examples/src/websocketsrv/AsrWebsocketServer.java]: Is there a reason to use a static member here?

File: java-api-examples/Makefile [hunk 2]
```diff
@@ -56,17 +71,32 @@ runmic:
 
 		java -cp ./lib/sherpaonnx.jar:build $(RUNJFLAGS)  DecodeMic
 
+runsrv:
```
Comment (csukuangfj) [inline java-api-examples/Makefile:75]: Could you document the message format/protocol for the communication?

File: java-api-examples/src/websocketsrv/AsrWebsocketServer.java (hunk 1 above)
Comment (zhaomingwork) [reply java-api-examples/src/websocketsrv/AsrWebsocketServer.java]: static maybe more efficiency and one srv instance assumption. I can remove it if needed

File: java-api-examples/Makefile (hunk 2 above)
Comment (zhaomingwork) [reply java-api-examples/Makefile:75]: ok

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**